rules/store/kv/store.go

```diff
@@ -96,7 +97,12 @@ func (s *store) WriteRuleSet(rs rules.MutableRuleSet) error {
 	}
 	conditions, ops := []kv.Condition{rsCond}, []kv.Op{rsOp}
 	_, err = s.kvStore.Commit(conditions, ops)
-	return err
+	switch err {
+	case kv.ErrConditionCheckFailed:
```

Comment:
I made this a switch in case we want to add some other cases in the future.

Reply:
Would suggest changing it to `err == kv.Err...`, can change to a switch if there are indeed more cases.

Do you need to add this to `WriteAll`? If so, maybe just add a `wrapWriteError` method and use it in both places.

Reply:
If adding to WriteAll, should also add the test case for it as well.

Reply:
@xichen2020 yeah I had that initially, they switched to this because I thought it was more extendible. I can switch back. 

I can also make the change to the WriteAll function.